api/client/service.proto

```diff
@@ -37,10 +39,40 @@ message ExecutionResponse {
   }
 }
 
+message RequestStreamRequest {
+  uint64 amount = 1;
+}
+
+message RequestSpecifier {
+  // method/path/uri/authority/content_length can all be expressed in request headers.
+  // so technically they are redundant. In case of ambiguity headers may end up being duplicated.
+  // Note that as of today, the target upstream/cluster that will be used to replay traffic to
+  // fixes the destination address/port
+  google.protobuf.StringValue method = 1;
+  google.protobuf.StringValue uri = 2; // XXX(oschaaf): path?
+  google.protobuf.StringValue authority = 3;
+  envoy.api.v2.core.HeaderMap headers = 4;
+  google.protobuf.UInt32Value content_length = 5;
+  // TODO(oschaaf): nice to have
+  // google.protobuf.StringValue sni_hostname = 10;
+}
+
+message Expectations {
+  google.protobuf.UInt32Value response_code = 1;
+  google.protobuf.UInt32Value content_length = 2;
```

Comment:
Please comment all proto fields here and elsewhere.

Reply:
Done